doc/metadata-server/README.md

```diff
@@ -0,0 +1,327 @@
+  insMetadata <- extractInstanceMetadata
+
+  pure { insMetadata, extMetadata }
+```
+
+## Protocol
+
+### Objectives
+
+- We want to avoid centralization of metadata registry.
+
+- We want to minimize blockchain resource usage in the default setup.
+
+- We want to use [existing or emerging Cardano standards](https://cips.cardano.org/) and tools.
+
+- Metadata format schema should be well defined but we should leave it open / extensible.
+
+### Marlowe metadatum label
+
+As a baseline channel for metadata exchange we want to use [`TxMetadata`](https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/tx-metadata.md) which simplified schema looks like this:
+
+```CDDL
+transaction_metadata =
+  { * transaction_metadatum_label => transaction_metadatum }
+```
+
+`transaction_metadatum_label` is a namespace which dApps should use to indicate the type and purpose of the transaction and its metadata. To properly use this standard we should reserve a place in an [existing label registry](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0010/registry.json) which is managed by Cardano Foundation. Let's assume that we use a single namespace `1564` (Kit Marlowe [1564-1593]) for Marlowe metadata exchange.
+
+### `transaction_metadatum`
+
+`transaction_metadatatum` can be nearly any `CBOR` encoded data (with string / byte buffers length limitations to `64B`). [Tx metadata specification](https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/tx-metadata.md#metadata-schemas---mappings-and-formats) defines a way to encode metadata using `cardano-cli` and `JSON` encoded payload. For the purpose of this draft we use simplified ["no schema"](https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/tx-metadata.md#no-schema) style spec (just plain `JSON`) of the format which is more readble and can be quickly tested with `cardano-cli`.
+We should expand this in the next iterations into a [detailed schema](https://github.com/input-output-hk/cardano-node/blob/master/doc/reference/tx-metadata.md#detailed-schema) which is translated into a precise CBOR representation (both `cardano-cli` and [`node-wallet`](https://input-output-hk.github.io/cardano-wallet/user-guide/common-use-cases/TxMetadata) support it). We want to use precise CBOR encoding because this exchange protocol should be tool and language agnostic.
+
+#### Storage
+
+Using blockchain as the only source of data has many advantages - we don't have to rely on things like hosting, domain registration, maintenance of the data storage systems etc. On the other hand, storing data on-chain has its cost because it increases transaction fees. `TxMetadata` size contributes to the overall transaction size (without any additional fees). Additionally transactions have currently [64KB](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0009/README.md#updatable-protocol-parameters) size limit which is an additional "risky" factor.
```

Comment:
Actually, it might increase the transaction fee. We should check experimentally.